python/golem_task_api/server.py

```diff
@@ -108,23 +76,74 @@ def __init__(self, work_dir: Path, handler: GolemAppHandler) -> None:
         reply.score = score
         await stream.send_message(reply)
 
+    async def Shutdown(self, stream):
+        _ = await stream.recv_message()
+        self._shutdown_future.set_result(None)
+        reply = ShutdownReply()
+        await stream.send_message(reply)
+
 
-class GolemAppServer:
+class RequestorAppServer:
     def __init__(
             self,
             work_dir: Path,
             port: int,
-            golem_app_handler: GolemAppHandler) -> None:
-        golem_app = GolemApp(work_dir, golem_app_handler)
+            handler: RequestorAppHandler,
+    ) -> None:
         loop = asyncio.get_event_loop()
+        self._shutdown_future = loop.create_future()
+        golem_app = RequestorApp(work_dir, handler, self._shutdown_future)
         self._server = Server(handlers=[golem_app], loop=loop)
         self._port = port
 
     async def start(self):
-        print(f'Starting server at port {self._port}')
         await self._server.start('', self._port, ssl=None)
+        print(f'Starting server at port {self._port}')
```

Comment:
Is the server is already `Started` at this point?

Reply:
Right, I think I moved that line while debugging